eng/pipelines/installer/jobs/base-job.yml

```diff
@@ -0,0 +1,173 @@
+parameters:
+  buildConfig: ''
+  osGroup: ''
+  archType: ''
+  osSubgroup: ''
+  timeoutInMinutes: 120
+  condition: true
+  container: ''
+  prepareSteps: []
+  buildSteps: []
+  dependsOn: []
+  variables: []
+  buildVariables: {}
+  name: ''
+  displayName: ''
+  pool: ''
+  workspace:
+    clean: all
+  strategy:
+    matrix:
+      release:
+        _BuildConfig: Release
+
+  buildCommandSourcesDirectory: '$(Build.SourcesDirectory)/'
+
+  liveCoreClrBuildConfig: ''
+  liveLibrariesBuildConfig: ''
+
+jobs:
+- job: ${{ format('installer_{0}', coalesce(parameters.name, parameters.platform)) }}
+  displayName: ${{ format('Installer Build and Test {0}', coalesce(parameters.name, parameters.platform)) }}
+
+  condition: and(succeeded(), ${{ parameters.condition }})
+  pool: ${{ parameters.pool }}
+  strategy: ${{ parameters.strategy }}
+  timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+
+  ${{ if ne(parameters.workspace, '') }}:
+    workspace: ${{ parameters.workspace }}
+
+  variables:
+  - ${{ parameters.variables }}
+  - ${{ each variable in parameters.buildVariables }}:
+    - name: ${{ variable.key }}
+      value: ${{ variable.value }}
+
+  # Produce test-signed build for PR and Public builds
+  - ${{ if or(eq(variables['System.TeamProject'], 'public'), in(variables['Build.Reason'], 'PullRequest')) }}:
+    - name: SignType
+      value: test
+
+  # Set up non-PR build from internal project
+  - ${{ if eq(variables.isOfficialBuild, true) }}:
+    - name: SignType
+      value: $[ coalesce(variables.OfficialSignType, 'real') ]
+
+  - name: LiveOverridePathArgs
+    value: >-
+      $(CoreCLROverridePathArg)
+      $(CoreFXOverridePathArg)
+
+  - name: CoreCLROverridePathArg
+    value: ''
+  - name: CoreFXOverridePathArg
+    value: ''
+
+  - name: CoreClrDownloadPath
+    value: ''
+  - name: LibrariesDownloadPath
+    value: ''
+
+  - ${{ if ne(parameters.liveCoreClrBuildConfig, '') }}:
+    - name: liveCoreClrLegName
```

Comment:
> I originally chickened out of Santi's suggestion to put this logic in xplat-setup (and you're not using it, after all)

I actually am using xplat-setup, since I use platform-matrix... or am I misunderstanding?

It makes sense to me to have it in xplat-setup, since the bulk of the transfer is setting up variables, and I think this is the only reasonable mechanism we have to share agent-local variable setup.

Reply:
> > I originally chickened out of Santi's suggestion to put this logic in xplat-setup (and you're not using it, after all)
> 
> I actually am using xplat-setup, since I use platform-matrix... or am I misunderstanding?

We really need to update our docs on how these templates flow. I'll open a tracking issue for that.

Reply:
The reasoning was to have all this `*ArtifactName` variables and `LegName` defined in `xplat-setup` so that all templates can benefit from it and just use them without having to actually define it in multiple places. Currently each partition defined this on it's base template.

However the change is not trivial as the right parameters need to be flown to the right templates in order to be able to access the values we need on `xplat-setup`. So this can be a follow up clean-up change. 

I still have to do clean-up on `libraries` template which I left from my previous PR to unblock work, so I can put it as part of my change.

Reply:
Please let's make sure we don't get deadlocked by these considerations for now - it's probably up to me to apologize for bringing them up in the first place. I'll be the happiest person in the world to clean this up but it's strictly ordered after standing up the official build and producing signing artifacts so please let's not block on that for now. I'll be happy to play with pipeline cleanups over / after the EOY holidays ;-).